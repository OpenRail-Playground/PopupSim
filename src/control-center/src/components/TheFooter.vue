<script setup lang="ts"></script>

<template>
  <footer class="rea-footer" role="contentinfo">
    <ul class="cmp-metanavigation">
      <li>
        <RouterLink to="/imprint" class="elm-link" :title="$t('imprint')">
          {{ $t('imprint') }}
        </RouterLink>
      </li>
      <li>
        <RouterLink to="/help" class="elm-link" :title="$t('help')">
          {{ $t('help') }}
        </RouterLink>
      </li>
    </ul>
    Copyright 2024
  </footer>
</template>

<style lang="scss" scoped>
footer.rea-footer {
  position: sticky;
}
</style>
