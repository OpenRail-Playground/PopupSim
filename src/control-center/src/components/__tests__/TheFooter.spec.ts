import { describe, test, expect } from 'vitest'
import { createI18n } from 'vue-i18n'
import { mount } from '@vue/test-utils'

import TheFooter from '../TheFooter.vue'

describe('TheFooter', () => {
  test('renders properly', () => {
    const i18n = createI18n({})
    const wrapper = mount(TheFooter, {
      global: {
        plugins: [i18n]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
