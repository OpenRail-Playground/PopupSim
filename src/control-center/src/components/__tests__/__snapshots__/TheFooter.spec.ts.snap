// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheFooter > renders properly 1`] = `
<footer
  class="rea-footer"
  data-v-49c9d9a5=""
  role="contentinfo"
>
  <ul
    class="cmp-metanavigation"
    data-v-49c9d9a5=""
  >
    <li
      data-v-49c9d9a5=""
    >
      <routerlink
        class="elm-link"
        data-v-49c9d9a5=""
        title="imprint"
        to="/imprint"
      >
        imprint
      </routerlink>
    </li>
    <li
      data-v-49c9d9a5=""
    >
      <routerlink
        class="elm-link"
        data-v-49c9d9a5=""
        title="help"
        to="/help"
      >
        help
      </routerlink>
    </li>
  </ul>
   Copyright 2022 DB Systel 
</footer>
`;
